api/search.go

```diff
@@ -299,12 +387,29 @@ func SearchHandlerFunc(validator QueryParamValidator, queryBuilder QueryBuilder,
 
 // LegacySearchHandlerFunc returns a http handler function handling search api requests.
 // TODO: This wil be deleted once the switch over is done to ES 7.10
-func LegacySearchHandlerFunc(validator QueryParamValidator, queryBuilder QueryBuilder, elasticSearchClient ElasticSearcher, transformer ResponseTransformer) http.HandlerFunc {
+func LegacySearchHandlerFunc(validator QueryParamValidator, queryBuilder QueryBuilder, nlpConfig config.NLP, cli *nlp.Client, elasticSearchClient ElasticSearcher, transformer ResponseTransformer) http.HandlerFunc {
 	return func(w http.ResponseWriter, req *http.Request) {
 		ctx := req.Context()
 		params := req.URL.Query()
 
-		q, searchReq, countReq := CreateRequests(w, req, validator)
+		var nlpCriteria *query.NlpCriteria
+		q := params.Get("q")
+		if params.Get("c") == "1" {
+			nlpSettings := query.NlpSettings{}
+
+			// Load default settings
+			// FIXME: move this somewhere better
+			json.Unmarshal([]byte(nlpConfig.NlpHubSettings), &nlpSettings)
+
+			// Load settings for this request
+			nlpSettingsRequest := params.Get("nlpSettings")
+			if nlpSettingsRequest != "" {
+				json.Unmarshal([]byte(nlpSettingsRequest), &nlpSettings)
```

Comment:
Are we OK taking the settings from the query or are there settings we should not? (are those all envs?)